File: ClientRuntimes/Ruby/ms-rest-azure/lib/ms_rest_azure.rb
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

require 'ms_rest'

require 'ms_rest_azure/version'

require 'ms_rest_azure/resource.rb'
require 'ms_rest_azure/sub_resource.rb'
require 'ms_rest_azure/cloud_error.rb'
require 'ms_rest_azure/azure_operation_response.rb'
require 'ms_rest_azure/async_operation_status.rb'
require 'ms_rest_azure/polling_state.rb'
require 'ms_rest_azure/azure_application_credentials.rb'
require 'ms_rest_azure/azure_service_client.rb'
require 'ms_rest_azure/token_refresh_middleware'

module MsRestAzure; end

File: ClientRuntimes/NodeJS/ms-rest-azure/lib/msRestAzure.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.
'use strict';

exports.AzureServiceClient = require('./azureServiceClient');
exports.UserTokenCredentials = require('./credentials/userTokenCredentials');
exports.ApplicationTokenCredentials = require('./credentials/applicationTokenCredentials');
exports.DeviceTokenCredentials = require('./credentials/deviceTokenCredentials');
exports.AzureEnvironment = require('./azureEnvironment');
exports.BaseResource = require('./baseResource');
exports.CloudError = require('./cloudError');
exports.TokenCredentials = require('ms-rest').TokenCredentials;

exports.generateUuid = require('./utils').generateUuid;
exports.interactiveLogin = require('./login').interactive;
exports.loginWithUsernamePassword = require('./login').withUsernamePassword;
exports.loginWithServicePrincipalSecret = require('./login').withServicePrincipalSecret;
exports = module.exports;